pkg/controller/master-controller-manager/rbac/mapper.go

```diff
@@ -646,13 +646,28 @@ func generateVerbsForNamespacedResource(groupName, resourceKind, namespace strin
 	// special case - only the owners of a project and project managers can create secrets in "saSecretsNamespaceName" namespace
 	//
 	if namespace == saSecretsNamespaceName || strings.HasPrefix(namespace, resources.KubeOneNamespacePrefix) {
-		switch {
-		case strings.HasPrefix(groupName, OwnerGroupNamePrefix) && resourceKind == secretV1Kind:
-			return []string{"create"}, nil
-		case strings.HasPrefix(groupName, ProjectManagerGroupNamePrefix) && resourceKind == secretV1Kind:
-			return []string{"create"}, nil
-		case resourceKind == secretV1Kind:
-			return nil, nil
+		if resourceKind == secretV1Kind {
+			switch {
+			case strings.HasPrefix(groupName, OwnerGroupNamePrefix):
+				return []string{"create"}, nil
+			case strings.HasPrefix(groupName, ProjectManagerGroupNamePrefix):
+				return []string{"create"}, nil
+			default:
+				return nil, nil
+			}
+		}
+
+		if resourceKind == kubermaticv1.ClusterBackupStorageLocationKind && namespace == resources.KubermaticNamespace {
+			switch {
+			case strings.HasPrefix(groupName, OwnerGroupNamePrefix):
+				return []string{"create"}, nil
+			case strings.HasPrefix(groupName, EditorGroupNamePrefix):
+				return []string{"create"}, nil
+			case strings.HasPrefix(groupName, ProjectManagerGroupNamePrefix):
+				return []string{"create"}, nil
+			default:
+				return nil, nil
+			}
 		}
 	}
```

Comment:
The outer `if` condition checks for `namespace == saSecretsNamespaceName` (which is `kubermatic`), and then the `if` for CBSL checks `namespace == resources.KubermaticNamespace` again. Should there maybe be a high-level `switch` condition on `resourceKind`?